### DISCUSSION TITLE
Another Plea

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#806] [Q&A] [answered]
This is a request to rename Location/Groups to simply Location and have whatever the admin entered in the location field display on the meeting list table. 

I've mentioned this a few times that displaying group name (if there is one) in place of the location for online meetings is very confusing and looks weird. We have multiple groups doing online and in-person meetings concurrently. These are not hybrid meetings, but rather separate meetings by the same group meeting at the same time, and often following the same topic. 

I believe I was told the logic for displaying groups was to encourage meetings to become groups. Yes, we should be educating people about the difference between a meeting and a group, but I don't believe that should be the mission of TSML. I've looked at dozens of websites using TSML, and all I see under Location/Groups heading for online meetings many empty cells.

I get that you don't want it to be ugly cluttered looking with crossed out lines where the address is crossed out, and maybe if any meeting is online, where attendance option is either online, or tc, then the word "online" can be inserted in the displayed location cell. 

A previous request was made to to change the text “Location Temporarily Closed” to “Not Meeting at Physical Location” or  “Meeting Online Only” or something like that, and I mentioned I put this request into appsupport@aa.org to change the Meeting Guide App, which is still pending. The response I got from you was that you'd sync this change when they make the change. Fair enough. 

But with respect to what is being displayed through the App, for natively online meetings, the location field displays what the admin entered. 

Below is from TSML. Wake Up Call is running two meetings at the same time. One is online and the other in-person. Both meetings are by the same group. Bethany Early Birds does a similar arrangement. Yet looking at this gives the impression that only one of these meetings are part of Wake Up Call and only one of these meetings are part of Early Birds.
![TSML UI Meetings Group](https://user-images.githubusercontent.com/62247421/168485440-367b0de0-89c0-40a7-8297-03e35c5a5ec6.png)

Here's how the App displays this.
![TSML UI Meetings GroupApp](https://user-images.githubusercontent.com/62247421/168485699-bf8f8240-f2b1-42b5-9c51-f9f16b0e1e01.png)

This is a natively online meeting. Below is TSML.
![Daily Reflections](https://user-images.githubusercontent.com/62247421/168485494-095850a1-dfbe-4ab3-8390-4920f65344db.png)

Same meeting on the App. Notice it's pulling from the local field.
![Daily Reflections App2](https://user-images.githubusercontent.com/62247421/168485675-8ce6a785-7987-4b42-be27-dd8a969bec6c.png)

**Comment by joshreisner** [comment, marked as answer]
hi @Grateful-Viv this is something you can actually do now in TSML UI by adding a custom script:

```
<script>
  var tsml_react_config = {
    columns: ['time', 'distance', 'name', 'location', 'address', 'region'],
  };
</script>
```

we're talking about it internally but currently it's TSML UI-only

**Reply by Grateful-Viv** [reply]
Where would I install the script? In the Head section?

**Reply by joshreisner** [reply]
yes that should work!

**Reply by Grateful-Viv** [reply]
It doesn't change anything. :(

**Reply by joshreisner** [reply]
sorry! i was testing it on a [non-TSML](https://tsml-ui.code4recovery.org/tests/location.html) site. i think we need to figure out the relationship to TSML to make it work properly there.

**Reply by Grateful-Viv** [reply]
App support responded they weren’t inclined to change the text saying location temporarily closed to and suggested for these meetings that have been online (now for over two years) be divorced from the location they once were and simply change it to online only or something like that in the location field. 

App support responded through they’re still discussing the possibility of updating some of the type labels, they have no plans to do anything in the near future. And "Meeting Online" would likely not be a direct replacement for "Temporarily Closed" as there are other reasons locations close temporarily. Suggested changing status to online only (by removing the actual street address) and entering in the location field “online” or something like that. 

I’m inclined to do that, and it would display on the app whatever I put in the location field such as “online meeting”, but TSML isn’t doing that — instead it’s either the group name or empty if there isn’t a group name. 

The suggestion was made by you to add “online” or whatever to the group field for these meetings, but that then changes group name for all the online only meetings to that new group name. Many of these online only meetings are real groups and I can’t have that over written. 

So once again, I’m hoping TSML will display what the admin entered in the location field for online meetings & not pull the group name if there is one or leave it blank if there isn’t one. This would be consistent with the AA meeting guide app. 

Thanks.

**Reply by joshreisner** [reply]
hi Viv, in the next release you will be able to add code to bring back the location-only column. this will work for TSML and TSML UI ✅

**Reply by Grateful-Viv** [reply]
@joshreisner I see the new release (v 3.14.4) is out. How would I go about changing the column to display only location information?

**Reply by joshreisner** [reply]
@Grateful-Viv try this: https://wordpress.org/plugins/12-step-meeting-list/#can%20i%20change%20the%20%22location%20%2F%20group%22%20column%20to%20display%20only%20the%20location%20name%20instead%3F

**Reply by Grateful-Viv** [reply]
So I’ll need a child. Thanks.

**Reply by joshreisner** [reply]
if it's easier for you i think you could add it to your `wp-config.php`

**Reply by Grateful-Viv** [reply]
Do you know if that would work? I suppose I could save a copy of the wp-config.php and try it. If it works, that's great. If it blows up, then I can restore the saved copy.

**Reply by joshreisner** [reply]
yes probably good to create a backup copy before making changes. but yes it only occurred to me recently that this should be possible -- it doesn't get touched by wordpress updates

**Reply by Grateful-Viv** [reply]
I finally added this to the wpconfig.php.  I tried to create a child, but ran into problems and reached out to the theme developer who said this should probably have been handled using the standard WordPress filter or action mechanism and that I didn't need a child theme. That got lost on me unfortunately. 

Anyway, adding it to the end of the wpconfig.php  changed the label from location/groups to location, but the data showing is still pulling from group data if there is any, not location data for TSML Legacy. It shows the correct location data for TSML-UI, however.  

Also region is now the left column, so I am moving that after Types and changing the label Name to Meeting. What is distance? I don't see that displayed?

```
$tsml_columns = array(
    'time' => 'Time',
    'distance' => 'Distance',
    'name' => 'Meeting',
    'location' => 'Location',
    'address' => 'Address',
    'types' => 'Types' ,
    'region' => 'Region'
);
```

Thank you.

**Reply by joshreisner** [reply]
The distance column only appears if you are searching for meetings near you or near a location. 

We should probably update that example to use the normal order with Region last.

Adding this code works for on our [TSML demo page](https://code4recovery.org/meetings). I am seeing the location data now displaying, no group data.

Are you running the latest version of TSML? The only other reason I can think of is if you had overridden the `archive-meetings.php` file in your theme, but that sounds unlikely.

**Reply by joshreisner** [reply]
The only other thing that occurs to me is caching, but if the column name changed for you then it doesn't sound like that's the issue.

**Reply by Grateful-Viv** [reply]
No changes to any PHP files.

How are you able to have same location names at multiple addresses? For example, Online Meeting at San Jose, Online Meeting at Santa Clara? 

I can't seem to make it stick for same location name, such as Online Meeting to have multiple regions.

Or if I have multiple location names using the same region, it will change all meetings using that region to the  same location name.

Also, is there any way through CSS if the attendance option is TC that I can display a line through the location to indicate this meeting is not meeting at the location?
